x-pack/legacy/plugins/apm/public/components/app/Settings/AgentConfigurations/AgentConfigurationCreateEdit/SettingsPage/saveConfig.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { NotificationsStart } from 'kibana/public';
+import { AgentConfigurationIntake } from '../../../../../../../../../../plugins/apm/common/runtime_types/agent_configuration/configuration_types';
+import {
+  getOptionLabel,
+  omitAllOption
+} from '../../../../../../../../../../plugins/apm/common/agent_configuration_constants';
+import { callApmApi } from '../../../../../../services/rest/createCallApmApi';
+
+export async function saveConfig({
+  config,
+  isEditMode,
+  toasts
+}: {
+  config: AgentConfigurationIntake;
+  agentName?: string;
+  isEditMode: boolean;
+  toasts: NotificationsStart['toasts'];
+}) {
+  try {
+    await callApmApi({
+      pathname: '/api/apm/settings/agent-configuration',
+      method: 'PUT',
+      params: {
+        query: { overwrite: isEditMode },
```

Comment:
Very cool, will do the same on saveCustomLink 👍🏻

Reply:
Only if it makes sense. For custom links you have an id as identifier so there is not a risk of overwriting by mistake, is there?